### DISCUSSION TITLE
Workflow context when triggered by issue_comment

### DISCUSSION THREAD

**Discussion by fernap3** [community/community#25917] [Actions]
<p>Hi!  I am writing a simple bot that will run status checks when a comment (something like “<span class="mention">@mybot</span> build”) is made on a pull request.</p>
<p>I notice that when a workflow is triggered by issue_comment as the result of a comment being made on a pull request, github.ref is always refs/heads/master.  What I really want is for the context to be the branch that the pull request is on so the status checks in the pull request will update accordingly.</p>
<p>Is this a bug?  If this is expected behavior, then I guess I will need to use the Statuses API explicitly to get the pull request to update properly in response to a check being run as the result of a comment.</p>
<p>Peter</p>

**Comment by ethomson** [comment, marked as answer]
<p>Comments on issues and pull requests will have the <code>ref</code> <a href="https://help.github.com/en/articles/events-that-trigger-workflows" rel="nofollow noopener">set to the default branch</a>.  However, you should be able to inspect the full <a href="https://help.github.com/en/articles/contexts-and-expression-syntax-for-github-actions#contexts" rel="nofollow noopener">github event context </a>for the information about the pull request in question.</p>
<p>I think that what you’re looking for is github.event.issue.pull_request and github.event.issue.number.</p>

**Comment by fernap3** [comment]
<p>Yeah, thanks for the reply.  I can indeed get the pull request number from the event, but I can’t get the PR ref directly from the event which means I need to do an API call to figure out what branch to checkout.  Complicates the workflow but it’s doable.</p>

**Comment by Kerizer** [comment]
<p>We have a similair issue. It would be really great to have something like</p>
<pre><code>github.event.pull_request
</code></pre>
<p> that containes all the information about pull request that was commented including head_ref and base_ref<br>
Performing additional api call inside workflow and parsing it is a bit messy imo.</p>

**Comment by carpenike** [comment]
<p>Trying to set this up myself now. Did you ever get it working?</p>

**Comment by Marvin9** [comment]
<p>This is what I tried,</p>
<pre><code>- uses: actions/checkout@v2
- run: |
       git fetch origin +refs/pull/${{ github.event.issue.number }}/merge
       git checkout FETCH_HEAD
</code></pre>

**Comment by timbrownsf** [comment]
<p>I’d suggest using this workflow package: <a href="https://github.com/Khan/pull-request-comment-trigger" rel="nofollow noopener">https://github.com/Khan/pull-request-comment-trigger</a></p>

**Comment by umermughal** [comment]
@ethomson So I always need to merge my changes in default branch first?
